messages/ruby/lib/cucumber/messages/message/utils.rb

```diff
@@ -0,0 +1,45 @@
+module Cucumber
+  module Messages
+    class Message
+      module Utils
+        module ClassMethods
+
+          ##
+          # Makes an underscored, lowercase form from the expression in the string.
+          #
+          #   underscore('GherkinDocument')         # => "gherkin_document"
+          #
+          # This is a simplified version of the Ruby on Rails implementation
+          # https://github.com/rails/rails/blob/v6.1.3.2/activesupport/lib/active_support/inflector/methods.rb#L92
+
+          def underscore(camel_cased_word)
+            return camel_cased_word unless /[A-Z-]/.match?(camel_cased_word)
+            word = camel_cased_word.gsub(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2')
+            word.gsub!(/([a-z\d])([A-Z])/, '\1_\2')
+            word.tr!("-", "_")
+            word.downcase!
+            word
+          end
+
+          ##
+          # Converts strings to UpperCamelCase.
+          #
+          #   camelize('gherkin_document')                # => "GherkinDocument"
+          #
+          # This is a simplified version of the Ruby on Rails implementation
+          # https://github.com/rails/rails/blob/v6.1.3.2/activesupport/lib/active_support/inflector/methods.rb#L69
+
+          def camelize(term)
+            camelized = term.to_s
+            camelized.gsub!(/(?:_|(\/))([a-z\d]*)/i) { "#{$1}#{$2.capitalize}" }
+            camelized
+          end
+        end
+
+        def self.included(other)
```

Comment:
traditionally this would be at the top as this is the first thing you read. Similar to what I have in SitePrism here: https://github.com/site-prism/site_prism/blob/main/lib/site_prism/dsl.rb#L11